Sources/MongoDriver.swift

```diff
@@ -2,103 +2,99 @@ import Fluent
-    */
-    public var idKey: String = "_id"
-
-    /**
-        Describes the types of errors
-        this driver can throw.
-    */
-    public enum Error: ErrorProtocol {
+     Describes the types of errors
+     this driver can throw.
+     */
+    public enum Error: Swift.Error {
         case unsupported(String)
     }
     
-    var database: MongoKitten.Database
-
+    let database: MongoKitten.Database
+    
     /**
-        Creates a new `MongoDriver` with
-        the given database name, credentials, and port. 
-    */
+     Creates a new `MongoDriver` with
+     the given database name, credentials, and port.
+     */
     public init(database: String, user: String, password: String, host: String, port: Int) throws {
         let server = try Server("mongodb://\(user):\(password)@\(host):\(port)", automatically: true)
         self.database = server[database]
     }
     
-    /**
-        Executes a query on the current MongoDB database.
-    */
-    public func execute<T: Model>(_ query: Fluent.Query<T>) throws -> [[String: Fluent.Value]] {
-        var items: [[String: Fluent.Value]] = []
+    // MARK: All the Driver protocol implementations
 
+    /**
+     MongoDB uses `_id` as the main identifier.
+     */
+    public var idKey: String = "_id"
+    
+    /**
+     Executes a query on the current MongoDB database.
+     */
+    public func query<T : Entity>(_ query: Fluent.Query<T>) throws -> Node {
         print("Mongo executing: \(query)")
-
+        
         switch query.action {
         case .fetch:
             let cursor = try select(query)
+            var items: [Node] = []
             for document in cursor {
-                let item = convert(document: document)
-                items.append(item)
+                let i = convert(document: document)
+                items.append(i)
             }
+            return try items.makeNode()
         case .create:
             let document = try insert(query)
             if let document = document {
-                let item = convert(document: document)
-                items.append(item)
+                return convert(document: document)
+            } else {
+                throw Error.unsupported("Couldn't create it I guess")
             }
         case .delete:
             try delete(query)
+            return Node.null
         default:
             throw Error.unsupported("Action \(query.action) is not yet supported.")
         }
-
-        return items
     }
-
-    /**
-        Provides a closure for executing raw, unsafe
-        queries on the Mongo database.
-    */
-    public func raw(closure: (MongoKitten.Database) -> Document) -> Document {
-        return closure(database)
+    
+    public func schema(_ schema: Schema) throws {
+        throw Error.unsupported("MONGO DOESNT HAVE SCHEMA RIGHT?")
+    }
+    
+    public func raw(_ raw: String, _ values: [Node]) throws -> Node {
+        throw Error.unsupported("Does MongoKitten allow this?")
```

Comment:
Nope, Mongo doesn't support this. Just throw something like `Error.unsupported("Mongo does not support raw queries")`